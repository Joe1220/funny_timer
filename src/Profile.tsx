import React from "react"

type IProfile = {
  username: string
  name: string
}

const Profile: React.FC<IProfile> = ({ username, name }) => {
  return (
    <div>
      <b>{username}</b>&nbsp;
      <span>({name})</span>
    </div>
  )
}

export default Profile
