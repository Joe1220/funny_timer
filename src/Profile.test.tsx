import React from "react"
import { render } from "@testing-library/react"
import Profile from "./Profile"

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const utils = render(<Profile username="jsh" name="조승현" />)
    expect(utils.container).toMatchSnapshot()
  })
  it("shows the props correctly", () => {
    const utils = render(<Profile username="jsh" name="조승현" />)
    utils.getByText("jsh") // velopert 라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText("(조승현)") // (조승현)) 이라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText(/조/) // 정규식 /조/ 을 통과하는 엘리먼트가 있는지 확인
  })
})
