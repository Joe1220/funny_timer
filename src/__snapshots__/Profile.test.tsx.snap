// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      jsh
    </b>
     
    <span>
      (
      조승현
      )
    </span>
  </div>
</div>
`;
